File: app/src/main/java/com/example/tree/zhihu/adapters/ShouyeAttentionAdapter.java
package com.example.tree.zhihu.adapters;

import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.tree.zhihu.activity.ColumnMessageActivity;
import com.example.tree.zhihu.gson.ColumnOverview;
import com.example.tree.zhihu.R;

import java.util.List;

public class ShouyeAttentionAdapter extends RecyclerView.Adapter<ShouyeAttentionAdapter.MyViewHolder> {

    Context context;
    ColumnOverview columnOverview;
    List<Bitmap> bitmaps;

    public ShouyeAttentionAdapter(Context context, ColumnOverview columnOverview, List<Bitmap> bitmaps) {
        this.context = context;
        this.columnOverview = columnOverview;
        this.bitmaps = bitmaps;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        return new MyViewHolder(LayoutInflater.from(context).inflate(R.layout.item_shouye_attention, viewGroup, false));
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder myViewHolder, final int i) {
        myViewHolder.text_tile.setText(columnOverview.getData().get(i).getName());
        myViewHolder.text_description.setText(columnOverview.getData().get(i).getDescription());
        if (bitmaps.size() > i) {
            myViewHolder.image_title.setImageBitmap(bitmaps.get(i));
        }
        myViewHolder.itemView.setOnClickListener(new View.OnClickListener(){

            @Override
            public void onClick(View v) {
                Intent intent = new Intent(context, ColumnMessageActivity.class);
                intent.putExtra("id",columnOverview.getData().get(i).getId());
                context.startActivity(intent);
            }
        });

    }


    @Override
    public int getItemCount() {
        return columnOverview.getData().size();
    }


    class MyViewHolder extends RecyclerView.ViewHolder {
        ImageView image_title;
        TextView text_tile;
        TextView text_description;

        MyViewHolder(@NonNull View itemView) {
            super(itemView);
            image_title = (ImageView) itemView.findViewById(R.id.picture_title_topic);
            text_tile = (TextView) itemView.findViewById(R.id.text_title_topic);
            text_description = (TextView) itemView.findViewById(R.id.text_description_topic);
        }
    }
}

File: app/src/main/java/com/example/tree/zhihu/adapters/ShouyeAttentionAdapter.kt
package com.example.tree.zhihu.adapters

import android.content.Context
import android.content.Intent
import android.graphics.Bitmap
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.example.tree.zhihu.activity.ColumnMessageActivity
import com.example.tree.zhihu.gson.ColumnOverview
import com.example.tree.zhihu.R

class ShouyeAttentionAdapter(internal var context: Context, internal var columnOverview: ColumnOverview, internal var bitmaps: List<Bitmap>) : RecyclerView.Adapter<ShouyeAttentionAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): MyViewHolder {
        return MyViewHolder(LayoutInflater.from(context).inflate(R.layout.item_shouye_attention, viewGroup, false))
    }

    override fun onBindViewHolder(myViewHolder: MyViewHolder, i: Int) {
        myViewHolder.text_tile.text = columnOverview.data[i].name
        myViewHolder.text_description.text = columnOverview.data[i].description
        if (bitmaps.size > i) {
            myViewHolder.image_title.setImageBitmap(bitmaps[i])
        }
        myViewHolder.itemView.setOnClickListener {
            val intent = Intent(context, ColumnMessageActivity::class.java)
            intent.putExtra("id", columnOverview.data[i].id)
            context.startActivity(intent)
        }

    }


    override fun getItemCount(): Int {
        return columnOverview.data.size
    }


    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var image_title: ImageView
        var text_tile: TextView
        var text_description: TextView

        init {
            image_title = itemView.findViewById<View>(R.id.picture_title_topic) as ImageView
            text_tile = itemView.findViewById<View>(R.id.text_title_topic) as TextView
            text_description = itemView.findViewById<View>(R.id.text_description_topic) as TextView
        }
    }
}
